src/bootstrap/src/ferrocene/install.rs

```diff
@@ -0,0 +1,279 @@
+            let destination = to.join(entry_relative);
+            let ty = entry.file_type();
+            if ty.is_dir() && !destination.exists() {
+                fs::create_dir_all(&destination)?;
+            } else if ty.is_file() {
+                fs::rename(entry.path(), destination).unwrap();
+            }
+        }
+        Ok(())
+    };
+
+    let sysconfdir = prefix.join(default_path(&builder.config.sysconfdir, "etc"));
+    remap(&image_dir.join("etc"), &sysconfdir).unwrap();
+
+    let datadir = prefix.join(default_path(&builder.config.datadir, "share"));
+    remap(&image_dir.join("share"), &datadir).unwrap();
+
+    let docdir = prefix.join(default_path(&builder.config.docdir, &format!("share/doc/{package}")));
+    remap(&image_dir.join("share/doc"), &docdir).unwrap();
+
+    let mandir = prefix.join(default_path(&builder.config.mandir, "share/man"));
+    remap(&image_dir.join("share/man"), &mandir).unwrap();
+
+    let libdir = prefix.join(default_path(&builder.config.libdir, "lib"));
+    remap(&image_dir.join("lib"), &libdir).unwrap();
+
+    let bindir = prefix.join(&builder.config.bindir); // Default in config.rs
+    remap(&image_dir.join("bin"), &bindir).unwrap();
+}
+
+fn default_path(config: &Option<PathBuf>, default: &str) -> PathBuf {
+    config.as_ref().cloned().unwrap_or_else(|| PathBuf::from(default))
+}
+
+macro_rules! install {
+    (($sel:ident, $builder:ident, $_config:ident),
+       $($name:ident,
+       $condition_name: ident = $path_or_alias: literal,
+       $default_cond:expr,
+       only_hosts: $only_hosts:expr,
+       $run_item:block $(, $c:ident)*;)+) => {
+        $(
+            #[derive(Debug, Clone, Hash, PartialEq, Eq)]
+        pub struct $name {
+            pub compiler: Compiler,
+            pub target: TargetSelection,
+        }
+
+        impl $name {
+            #[allow(dead_code)]
+            fn should_build(config: &Config) -> bool {
+                config.extended && config.tools.as_ref()
+                    .map_or(true, |t| t.contains($path_or_alias))
+            }
+        }
+
+        impl Step for $name {
+            type Output = ();
+            const DEFAULT: bool = true;
+            const ONLY_HOSTS: bool = $only_hosts;
+            $(const $c: bool = true;)*
+
+            fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+                let $_config = &run.builder.config;
+                run.$condition_name($path_or_alias).default_condition($default_cond)
+            }
+
+            fn make_run(run: RunConfig<'_>) {
+                run.builder.ensure($name {
+                    compiler: run.builder.compiler(run.builder.top_stage, run.builder.config.build),
+                    target: run.target,
+                });
+            }
+
+            fn run($sel, $builder: &Builder<'_>) {
+                $run_item
+            }
+        })+
+    }
+}
+
+install!((self, builder, _config),
```

Comment:
If we want to provide support, we should replace this with the list of components we want people to actually be able to install.

Reply:
Oh, ya! This should have like `flip-link` etc. Are there any I should remove? 🤔

Reply:
We should probably remove the upstream docs.